server/memphis_handlers_tags.go

```diff
@@ -97,11 +92,10 @@ func AddTagsToEntity(tags []models.CreateTag, entity_type string, entity_id prim
 				return err
 			}
 			if !exist {
-				err = CreateTag(name, entity_type, entity_id, tag.Color)
+				err = CreateTag(name, entity, entity_id, tagToCreate.Color)
 				if err != nil {
 					return err
 				}
-				return nil
 			}
 			_, err = tagsCollection.UpdateOne(context.TODO(), bson.M{"_id": tag.ID}, bson.M{"$addToSet": bson.M{"stations": entity_id}})
```

Comment:
You can use updateOne with upsert intead of calling isExist & createTag.

After changing please ensure everything still works as expected and remove createTag function/isExist function in case no other functions use them